### PR TITLE
Incremental rebalancing

### PR DESCRIPTION
**Pull request by edenhill**
This can't be merged until there's a v1.6.0 librdkafka release (to be bundled), but I think it is good to have the review out of the way as soon as possible.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,24 @@
 # Confluent's Golang client for Apache Kafka
 
+## v1.6.0
+
+v1.6.0 is a feature release adding support for KIP-429 Incremental rebalancing.
```
Comment (mhowlett) [inline CHANGELOG.md:5]: also sticky partitioner.

Comment (mhowlett) [reply CHANGELOG.md:5]: very sticky release

File: kafka/consumer.go [hunk 2]
```diff
@@ -122,6 +121,63 @@ func (c *Consumer) Unassign() (err error) {
 	return nil
 }
 
+// IncrementalAssign adds the specified partitions to the current set of
+// partitions to consume.
+// The new partitions must not be part of the current assignment.
+func (c *Consumer) IncrementalAssign(partitions []TopicPartition) (err error) {
```
Comment (mhowlett) [inline kafka/consumer.go:127]: I assume you can specify offsets here? there's no comment about this and it doesn't follow from the typing.

File: kafka/consumer_test.go [hunk 3]
```diff
@@ -115,6 +115,38 @@ func TestConsumerAPIs(t *testing.T) {
 		t.Errorf("Unassign failed: %s", err)
 	}
 
+	// Incremental Assign & Unassign
+	err = c.IncrementalAssign([]TopicPartition{
+		{Topic: &topic1, Partition: 9},
```
Comment (mhowlett) [inline kafka/consumer_test.go:120]: offsets?

File: kafka/integration_test.go [hunk 4]
```diff
@@ -337,12 +338,16 @@ func consumerTest(t *testing.T, testname string, msgcnt int, cc consumerCtrl, co
...
 	conf := ConfigMap{"bootstrap.servers": testconf.Brokers,
 		"go.events.channel.enable": cc.useChannel,
-		"group.id":                 testconf.GroupID,
-		"session.timeout.ms":       6000,
-		"api.version.request":      "true",
-		"enable.auto.commit":       cc.autoCommit,
-		"debug":                    ",",
-		"auto.offset.reset":        "earliest"}
+		"group.id": testconf.GroupID +
+			fmt.Sprintf("-%d", rand.Intn(1000000)),
```
Comment (mhowlett) [inline kafka/integration_test.go:342]: nit: a 1/Uuid chance of dying would be better

**Review by mhowlett — APPROVED** [review]

File: kafka/integration_test.go (hunk 4 above)
Comment (edenhill) [reply kafka/integration_test.go:342]: There's no uuid generator in the standard library, don't want to pull in an external dependency just for this.

**Comment by huanggx-sea** [comment]
Hi, any update for this? Seems like the librdkafka has already released v1.6.0.

**Pull request merged**